src/index.js

```diff
@@ -0,0 +1,89 @@
+const Lang = imports.lang;
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const Xapian = imports.gi.Xapian;
+
+const SEQUENCE_NUMBER_VALUE_NO = 0;
+const EXACT_TITLE_PREFIX = 'XEXACTS';
+const TITLE_PREFIX = 'S';
+const TAG_PREFIX = 'K';
+const ID_PREFIX = 'Q';
+
+const DEFAULT_WEIGHT = 1;
+const EXACT_WEIGHT = 27;
+
+const Index = new Lang.Class({
+    Name: 'Index',
+
+    _init: function () {
+        let db_dir = GLib.dir_make_tmp('db_dir_XXXXXX');
+        let stopwords = [];
+        let prefixes = {
+            'prefixes': [
+                {
+                    'field':  'exact_title',
+                    'prefix': EXACT_TITLE_PREFIX
+                },
+                {
+                    'field':  'title',
+                    'prefix': TITLE_PREFIX
+                }
+            ],
+            'booleanPrefixes': [
+                {
+                    'field': 'tag',
+                    'prefix': TAG_PREFIX
+                },
+                {
+                    'field': 'id',
+                    'prefix': ID_PREFIX
+                }
+            ]
+        };
+
+        this._db = new Xapian.WritableDatabase({ path: db_dir,
+                                                 action: Xapian.DatabaseAction.CREATE_OR_OVERWRITE,
+                                                 flags: Xapian.DatabaseFlags.BACKEND_GLASS });
+        this._db.init(null);
+        this._db.set_metadata('XbPrefixes', JSON.stringify(prefixes));
+        this._db.set_metadata('XbStopwords', JSON.stringify(stopwords));
```

Comment:
Our back-end seems to expect that field to be present, so I rather leave it there, even if its empty for now.